src/fortuno/checkers/int_template.inc

```diff
@@ -162,6 +166,7 @@
     logical, allocatable :: match(:,:)
     integer(i64) :: mismatchloc(2)
 
+    allocate(match(size(value1, dim=1), size(value1, dim=2)))
     match = value1 == value2
```

Comment:
```suggestion
    allocate(match(size(value1, dim=1), size(value1, dim=2)))
    match(:,:) = value1 == value2
```
Whenever lhs reallocation is not wanted (because lhs had been already allocated) or not possible (because lhs is not allocatable), the placeholder indices should be used for arrays on the lhs. This makes the intention of *assigning into* more pronounced.

Reply:
Doesn't this only occur when the LHS is an allocatable variable being assigned a value whose shape/length doesn’t match its current allocation (like, the LHS not being allocated)? In this case, a shape mismatch is impossible because you have if-statements guarding against it.

But happy to make the change